projects/angular/src/data/datagrid/datagrid-detail.ts

```diff
@@ -38,8 +39,27 @@ import { DetailService } from './providers/detail.service';
 export class ClrDatagridDetail {
   @ContentChild(ClrDatagridDetailHeader) header: ClrDatagridDetailHeader;
 
+  @Input('clrDetailAriaLabelledBy') ariaLabelledBy: string;
+  @Input('clrDetailAriaLabel') ariaLabel: string;
+
   constructor(public detailService: DetailService, public commonStrings: ClrCommonStringsService) {}
 
+  get labelledBy(): string {
+    if (this.ariaLabelledBy) {
+      return this.header ? this.header.titleId + ' ' + this.ariaLabelledBy : this.ariaLabelledBy;
+    } else {
+      return this.ariaLabel ? null : this.header ? this.header.titleId : '';
```

Comment:
```suggestion
      return this.ariaLabel || this.header?.titleId || '';
```

This is much easier to read than a nested ternary expression.

Reply:
Also, `this.ariaLabel` is not valid for `aria-label` because it is not a list of element IDs.

Reply:
Oh, I misread the code.